cirq-google/cirq_google/engine/engine_client.py

```diff
@@ -86,6 +86,12 @@ def submit(self, func: Callable[..., Awaitable[_R]], *args, **kw) -> duet.Awaita
         future = asyncio.run_coroutine_threadsafe(func(*args, **kw), self.loop)
         return duet.AwaitableFuture.wrap(future)
 
+    _instance = None
+
+    @classmethod
+    def instance(cls):
+        return AsyncioExecutor._instance if AsyncioExecutor._instance else cls()
```

Comment:
Should assign to `_instance` here to ensure this gets cached:
```suggestion
        if cls._instance is None:
            cls._instance = cls()
        return cls._instance
```